content/guides/autodiscovery.md

```diff
@@ -4,37 +4,75 @@ kind: guide
+
+As the Agent inspects each template and its service identifiers, it builds a mapping of service identifiers to template YAML (or JSON). Then, it inspects every running Docker container on the Agent's host, looking for containers that match any of the service identifiers (keys) in the identifier-to-template mapping. For each match, the Agent generates a static check configuration by substituting the matching container's IP address and port into the corresponding template. Then it enables the check.
+
+The Agent watches for Docker events—container creation, destruction, starts, and stops—and recompiles check configurations from the templates on such events. Any time Autodiscovery finds another instance of the some service it had previously generated configuration for—whether it's an additional container or the previous one redeployed—it builds and enables another static configuration.
+
+There are a few caveats:
+
+- The Agent caches templates it gets from KV stores. Changes to KV store templates require a restart of the Agent.
+- something about precedence of templates when multiple sources provided?
+- others?
+
+---
+
+The next section clarifies this abstract description of Autodiscovery with some examples.
+
+# How to set it up
+
+No matter what container orchestration platform you use, you'll first need to run a single [docker-dd-agent container](https://hub.docker.com/r/datadog/docker-dd-agent/) on every host in your cluster. Then you'll write check templates, storing them in one or more template sources.
+
+## Running the Agent Container
+
+If you use Kubernetes, see the [Kubernetes integration page](http://docs.datadoghq.com/integrations/kubernetes/#installation) for instructions on running docker-dd-agent. If you use Amazon ECS, see [its integration page](http://docs.datadoghq.com/integrations/ecs/#installation).
+
+If you use Docker Swarm, run the following command on one of your manager nodes:
 
     docker service create \
       --name dd-agent \
       --mode global \
       --mount type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock \
       --mount type=bind,source=/proc/,target=/host/proc/,ro=true \
       --mount type=bind,source=/sys/fs/cgroup/,target=/host/sys/fs/cgroup,ro=true \
-      -e API_KEY=<YOUR API KEY> \
+      -e API_KEY=<YOUR_DATADOG_API_KEY> \
       -e SD_BACKEND=docker \
       datadog/docker-dd-agent:latest
 
-For Kubernetes, you can follow our [Kubernetes Integration](http://docs.datadoghq.com/integrations/kubernetes/) to create a DaemonSet. We also have [Amazon ECS integration instructions](http://docs.datadoghq.com/integrations/ecs/) available.
+Otherwise, see the docker-dd-agent documentation for detailed instructions and a comprehensive list of supported environment variables.
```

Comment:
That's a good question. Presumably any orchestration platform that runs Docker containers?